From 9030eb81823e46eb374b52fec2e65fff2bb2dcf9 Mon Sep 17 00:00:00 2001
From: Gerry Kopec <Gerry.Kopec@windriver.com>
Date: Fri, 15 Feb 2019 16:15:37 -0500
Subject: [PATCH] Stein: Nova console address config optionality

Introduce option to nova to enable/disable the use of the vnc or spice
server proxyclient address found by the console compute init container.
This can be used to prevent the case where the found address overrides
what has already been defined in nova.conf by per host nova compute
daemonset overrides.
---
 nova/templates/bin/_nova-compute.sh.tpl | 2 ++
 nova/values.yaml                        | 1 +
 2 files changed, 3 insertions(+)

diff --git a/nova/templates/bin/_nova-compute.sh.tpl b/nova/templates/bin/_nova-compute.sh.tpl
index 84596a5..b3bcca7 100644
--- a/nova/templates/bin/_nova-compute.sh.tpl
+++ b/nova/templates/bin/_nova-compute.sh.tpl
@@ -20,5 +20,7 @@ set -ex
 
 exec nova-compute \
       --config-file /etc/nova/nova.conf \
+{{- if .Values.console.address_search_enabled }}
       --config-file /tmp/pod-shared/nova-console.conf \
+{{- end }}
       --config-file /tmp/pod-shared/nova-libvirt.conf
diff --git a/nova/values.yaml b/nova/values.yaml
index 9646ded..c0ec7fe 100644
--- a/nova/values.yaml
+++ b/nova/values.yaml
@@ -438,6 +438,7 @@ console:
     vncproxy:
       # IF blank, search default routing interface
       vncserver_proxyclient_interface:
+  address_search_enabled: true
 
 ssh:
   key_types:
-- 
1.8.3.1

